### DISCUSSION TITLE
Cannot Add Assets to Maps because "Level is Locked"

### DISCUSSION THREAD

**Discussion by kef-bassam** [carla-simulator/carla#6159] [Q&A]
Hi,
### Overview
I'm trying to edit some of the default Carla maps, or more specifically, copies of the default Carla maps by adding and positioning new assets within them (i.e. adding more poles, adding stationary cars etc.).

### System
I am using Ubuntu 20.04 and followed the [instructions in the docs](https://carla.readthedocs.io/en/latest/build_linux/) to successfully build Carla 0.9.14 from source on my machine.

### Setup
In my newly built carla, I used `make launch` to open the Carla project in Unreal Engine 4.26 and made copies of each of the optimized versions of the maps (Town01_Opt, Town02_Opt, etc.) using the Unreal Engine content browser.

### Problem
For some reason, I can drag-and-drop assets just fine in the copies of `Town01_Opt` and `Town10_Opt`. However, for the rest of the optimized maps I cannot. I cannot even select individual assets within the Unreal Engine view ports. The terminal gives the following message when I try to drag-and-drop assets:
> `LogEditorSelectUtils: Warning: SelectActor: The requested operation could not be completed because the level is locked.`

The following appears in a pop-up in the Unreal UI when I try to drag and drop assets: `The requested operation could not be completed because the level is locked.`

### Question
Any ideas on how to unlock these levels or the assets within those levels? I tried looking on Unreal forums but couldn't find anything related.

### Post-Script
P.S. I can add assets just fine to the unoptimized maps and I can use them as a last result however the `_Opt` versions obviously run faster and I'd like to use them if possible.

**Comment by ly2856366242** [comment]
我找到了对应的解决方案，你可以参考这个链接，里面有解决方案  https://carla.readthedocs.io/en/latest/tuto_M_custom_layers/
